code/core/src/main/java/com/adobe/marketing/mobile/util/CloneFailedException.java

```diff
@@ -11,18 +11,58 @@
 
 package com.adobe.marketing.mobile.util;
 
+import androidx.annotation.NonNull;
+
 /**
  * Exception thrown by {@link EventDataUtils} to indicate that an exception occurred during deep
  * clone.
  */
 public class CloneFailedException extends Exception {
 
+    private final Reason reason;
+
+    /** Enum to indicate the reason for the exception. */
+    enum Reason {
+        MAX_DEPTH_REACHED,
+        UNSUPPORTED_TYPE,
+        UNKNOWN
+    }
+
     /**
      * Constructor.
      *
      * @param message {@code String} message for the exception
      */
     public CloneFailedException(final String message) {
+        this(message, Reason.UNKNOWN);
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param reason the {@link Reason} for the exception
+     */
+    public CloneFailedException(@NonNull final Reason reason) {
+        this(reason.toString(), reason);
+    }
+
+    /**
+     * Private constructor to unify public constructors and allow cascading.
+     *
+     * @param message {@code String} message for the exception
+     * @param reason the {@link Reason} for the exception
+     */
+    private CloneFailedException(@NonNull final String message, @NonNull final Reason reason) {
         super(message);
+        this.reason = reason;
+    }
+
+    /**
+     * Returns the {@link Reason} for the exception.
+     *
+     * @return rteturns the {@link Reason} for the exception
+     */
+    @NonNull public Reason getReason() {
```

Comment:
Good point. I will make Reason nonpublic for now as this is only used for distinction within the EventData utils and the consumer of this exception does not need the reason as yet (because the only time this is thrown is when Max depth is reached)